### PR TITLE
Add mkdocs tabs option to math docs

### PR DESCRIPTION
**Pull request by brynpickering**
Improves math docs to include the math YAML snippet as a tab next to the LaTeX math string.

Summary of changes in this pull request:

* Add `mkdocs_tabbed` option to the math documentation writer.
* Change AttrDict YAML dumping so that it doesn't sort dict keys by default.

Example:

https://github.com/calliope-project/calliope/assets/17178478/a8095c1b-9608-4953-8c9d-ab7d358f8510

Reviewer checklist:

- [ ] Test(s) added to cover contribution
- [ ] Documentation updated
- [ ] Changelog updated
- [ ] Coverage maintained or improved

### CODE REVIEW

File: docs/pre_defined_math/index.md
Comment (irm-codebase) [inline docs/pre_defined_math/index.md:1]: This note is broken. Missing a `>` for the line below?

File: src/calliope/backend/latex_backend_model.py
Comment (irm-codebase) [inline src/calliope/backend/latex_backend_model.py:1]: Unrelated question: do these `noqa: F811` do anything? They do not seem to anger the linter on my machine.

Comment (brynpickering) [reply src/calliope/backend/latex_backend_model.py:1]: Good to know. An update to Ruff must have dealt with this.

File: docs/pre_defined_math/index.md
Comment (brynpickering) [reply docs/pre_defined_math/index.md:1]: Good catch - missing sufficient indentation. Fixing the YAML snippets to indent to 2 spaces taught my IDE that markdown files should be indented 2 spaces, but actually things only work with 4-space indents....

**Review by irm-codebase — DISMISSED** [review]
Looks great!

**Review by sjpfenninger — APPROVED** [review]
Exactly what I had in mind!

**Pull request merged**